Switch to new data API

ovrheat opened the pull request:
I own one of these, that's why I added data API support. I haven't changed anything about output, included raw hex data output (it was called 'Received data') field that was already present in the module.
Should we leave it as it was originally (about showing also raw data) or not? You decide :)

merbanan commented on src/devices/generic_temperature_sensor.c:
```diff
@@ -5,16 +5,25 @@
  * (at your option) any later version.
+ *
+ * Switched to new data API by 'ovrheat' Nik
  */
```
I am not so found of these type of changes. This info can be found in the git repository log. If you feel that you contributed significantly then add your name as a copyright line.

merbanan commented on src/devices/generic_temperature_sensor.c:
```diff
@@ -29,21 +38,41 @@ static int generic_temperature_sensor_callback(bitbuffer_t *bitbuffer) {
+	//fprintf(stdout, "Temp          = %f\n",fTemp);
+	//fprintf(stdout, "Model         = Generic temperature sensor 1\n");
+	//fprintf(stdout, "Received Data = %02x %02x %02x\n", bb[1][0], bb[1][1], bb[1][2]);
+	
```
Remove all these lines.

merbanan commented on src/devices/generic_temperature_sensor.c:
```diff
@@ -29,21 +38,41 @@ static int generic_temperature_sensor_callback(bitbuffer_t *bitbuffer) {
+                     "temperature_C",	"Temperature",	DATA_FORMAT, 	"%.02f C",	DATA_DOUBLE,	fTemp,
+                     "battery",      	"Battery?",		DATA_INT,    								battery,
+					 "rawdata",			"Raw data",		DATA_STRING,								received_data,
+                     NULL);
```
Reindent these lines so they are symetrical,

merbanan commented on src/devices/generic_temperature_sensor.c:
```diff
@@ -29,21 +38,41 @@ static int generic_temperature_sensor_callback(bitbuffer_t *bitbuffer) {
+		             "id",          	"Id",			DATA_FORMAT,	"\t %d",	DATA_INT,		device,
+                     "temperature_C",	"Temperature",	DATA_FORMAT, 	"%.02f C",	DATA_DOUBLE,	fTemp,
+                     "battery",      	"Battery?",		DATA_INT,    								battery,
+					 "rawdata",			"Raw data",		DATA_STRING,								received_data,
```
Remove the rawdata member.

merbanan commented:
Fix the comments and the code is ready for merging.

ovrheat commented:
All done, it should be ok now. I removed also the part which was printing raw output (included the buffer), and also removed my name. No time to mess with copyright laws today :smile:
